node/src/store.rs

```diff
@@ -202,14 +182,42 @@ pub enum StoreError {
     /// Failed to open the store.
     #[error("Error opening store: {0}")]
     OpenFailed(String),
+}
 
-    /// Invalid range of headers provided.
-    #[error("Invalid headers range")]
-    InvalidHeadersRange,
+/// Store insersion non-fatal errors.
+#[derive(Error, Debug)]
+pub enum StoreInsertionError {
+    /// Provided headers failed verification.
+    #[error("Provided headers failed verification: {0}")]
+    HeadersVerificationFailed(String),
+
+    /// Provided headers cannot be appended on existing headers of the store.
+    #[error("Provided headers failed to be verified with existing neighbors: {0}")]
+    NeighborsVerificationFailed(String),
+
+    /// Store containts are not met.
+    #[error("Contraints not met: {0}")]
+    ContraintsNotMet(BlockRangesError),
+
+    // TODO: Same hash for two different heights is not really possible
+    // and `ExtendedHeader::validate` would return an error.
+    // Remove this when a type-safe validation is implemented.
+    /// Hash already exists in the store.
+    #[error("Hash {0} already exists in store")]
+    HashExists(Hash),
+}
```

Comment:
Should I merge it back to `StoreError`? When I created it I wanted to make it easier to identify non-fatal insertion errors.